test/Harness.hs

```diff
@@ -47,10 +47,17 @@ equateDI dat1 dat2 =
      check "datatypeVars len"      (length . datatypeVars)      dat1 dat2
      check "datatypeInstTypes len" (length . datatypeInstTypes) dat1 dat2
      check "datatypeVariant"       datatypeVariant              dat1 dat2
-     check "datatypeCons len"      (length . datatypeCons)      dat1 dat2
-
      let sub = Map.fromList (zip (freeVariables (bndrParams (datatypeVars dat2)))
                                  (map VarT (freeVariables (bndrParams (datatypeVars dat1)))))
+#if MIN_VERSION_template_haskell(2,16,0)
+     -- datatypeReturnKind returns an error on earlier versions
```

Comment:
Similarly, this CPP might not be required if we are no longer `error`ing on older GHC versions.